### ISSUE TITLE
Unknown Fighting Art causing misc. API errors

### ISSUE THREAD

**Issue by toconnell** [toconnell/kdm-manager#520]
> User OID: 57f6262d8740d93072e9e6ad
> Method: GET
> URL: http://api.thewatcher.io/settlement/get_campaign/57f7bb4e8740d971bfb4acf2
> JSON: None
> 
```
Traceback (most recent call last):
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1639, in full_dispatch_request
 rv = self.dispatch_request()
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1625, in dispatch_request
 return self.view_functions[rule.endpoint](**req.view_args)
File "/home/toconnell/kdm-manager/v2/api/utils.py", line 516, in wrapped_function
 resp = make_response(f(*args, **kwargs))
File "/home/toconnell/kdm-manager/v2/api/api.py", line 262, in collection_action
 return asset_object.request_response(action)
File "/home/toconnell/kdm-manager/v2/api/models/settlements.py", line 3994, in request_response
 return Response(response=self.serialize('campaign'), status=200, mimetype="application/json")
File "/home/toconnell/kdm-manager/v2/api/models/settlements.py", line 460, in serialize
 output['game_assets']['inspirational_statue_options'] = self.get_available_fighting_arts(exclude_dead_survivors=False, return_type='JSON')
File "/home/toconnell/kdm-manager/v2/api/models/settlements.py", line 2213, in get_available_fighting_arts
 if fa_dict.get('sub_type', None) == 'secret_fighting_art':
AttributeError: 'NoneType' object has no attribute 'get'
```

**Comment by toconnell** [comment]
Sadly the error doesn't include the FA handle that's coming back None.

Here's the production settlement referenced in the error:
```
toconnell@ubuntu-trusty-1:~/kdm-manager/v2/api$ ./admin.py -S 57f7bb4e8740d971bfb4acf2
  _id                 57f7bb4e8740d971bfb4acf2                <class 'bson.objectid.ObjectId'>
  name                FUCKBLAKE                               <type 'unicode'>
  campaign            people_of_the_lantern                   <type 'unicode'>
  expansions          []                                      <type 'list'>
  created_by          57f62cf18740d9361f63e882                <class 'bson.objectid.ObjectId'>
  created_on          2016-10-07 15:12:14.503000              <type 'datetime.datetime'>
```

**Comment by toconnell** [comment]
Pulled the user, reproduced the error and the settlement in question had old (and I mean really old) survivors whose `fighting_arts` (and other) dicts used names instead of handles: the survivors are not normalized when the settlement is serialized, so on the settlement's _serliaze()_ call, we tried to `get_available_fighting_arts()` and ended up feeding the `get_asset()` method a name instead of a handle.

tl;dr the fix is to let the method back off to a name:
```
@@ -2267,8 +2269,12 @@ class Settlement(Models.UserAsset):

         # 3.) now, initialize our master set and remove the SFA's
         for fa_handle in fa_handles:
-            fa_dict = self.FightingArts.get_asset(fa_handle, raise_exception_if_not_found=False)
-            if fa_dict.get('sub_type', None) == 'secret_fighting_art':
+            fa_dict = self.FightingArts.get_asset(
+                fa_handle,
+                backoff_to_name=True,
+                raise_exception_if_not_found=False
+            )
+            if fa_dict is not None and fa_dict.get('sub_type', None) == 'secret_fighting_art':
                 fa_handles.remove
```

The fix for this will go out in the next release of the APi.
